tests/model_serving/maas_billing/body_base_routing/pre_auth_model_header/test_bbr_pre_auth_inference.py

```
@@ -0,0 +1,96 @@
+    @pytest.mark.tier1
+    def test_inference_model_in_body_only_invalid_key_returns_401(
+        self: Self,
+        request_session_http: requests.Session,
+        bbr_inference_url: str,
+        bbr_chat_payload: dict[str, Any],
+    ) -> None:
+        """Verify the gateway rejects inference with 401 when an invalid API key is used on the BBR endpoint."""
+        assert_bbr_inference_status(
+            session=request_session_http,
+            inference_url=bbr_inference_url,
+            headers=build_maas_headers(token="invalid-key-that-does-not-exist"),
```

Ruff (0.15.18)

[error] Possible hardcoded password assigned to argument: "token"

(S106)